models/Document/DocType/Dao.php

```diff
@@ -56,6 +74,15 @@ public function getById($id = null)
      */
     public function save()
     {
+        if (!$this->model->getId()) {
+            $listing = new Listing();
+            $listing = $listing->load();
+            $listing = array_map(function(Model\Document\DocType $item) {
+                return $item->getId();
+            }, $listing);
+            $id = $this->getNextId($listing);
+            $this->model->setId($id);
+        }
```

Comment:
Is this maybe something we can make generic in the `AutoIncrementTrait` by providing the listing class? 
```php
$this->getNextId(Listing::class);
```